world/petitions/petitions_commands.py

```diff
@@ -7,9 +7,204 @@
 from commands.mixins import RewardRPToolUseMixin
 from server.utils.exceptions import PayError, CommandError
 from server.utils.prettytable import PrettyTable
+from evennia.utils import evtable
 from world.petitions.forms import PetitionForm
 from world.petitions.exceptions import PetitionError
-from world.petitions.models import BrokeredSale, Petition
+from world.petitions.models import BrokeredSale, Petition, WantedAd
+from web.character.models import (Clue, SearchTag)
+
+
+class CmdMatchmaker(ArxCommand):
+    """
+    Creates or searches wanted ads
+
+        Usage:
+    -Viewing:
+        matchmaker
+        matchmaker/all
+    -Editing
+        matchmaker/active #Toggles on/off
+        matchmaker/blurb <text>
+        matchmaker/playtimes <start>,<end> #Assume EST
+        matchmaker/clues <clue1>,<clue2>
+        matchmaker/tags <tag1>,<tag2>
+        matchmaker/seeking <marriage>,<plot>,<protege>,<sponsor> #any
+    """
+    key = "matchmaker"
+    help_category = "social"
+    aliases = ["mm"]
+
+    def display_matches(self):
+        my_ad = self.caller.dompc.wanted_ad
+        ads = WantedAd.objects.filter(active=True)
+        matches_list = []
+        matched_list = []
+        for ad in ads:
+            matches = ad.match(self.caller.dompc)
+            if matches and (ad.TIME in matches):
+                matches_list.append(matches)
+                matched_list.append(ad)
+        table = evtable.EvTable(
+            "Name/Gender/Age",
+            "Matches",
+            "Times",
+            "Blurb",
+            border="cells",
+            width=78)
+        if len(matches_list) > 0 and len(matched_list):
+            for match, ad in zip(matches_list, matched_list):
+                matchstring = ""
+                for m in match:
+                    matchstring += "[{}]".format(ad.MATCH_TYPES[m][1])
+                table.add_row("{}/{}/{}".format(ad.owner.player.char_ob.key,
+                                                ad.owner.player.char_ob.db.gender[0],
+                                                ad.owner.player.char_ob.db.age),
+                              matchstring,
+                              "{}-{}".format(ad.playtime_start,
+                                             ad.playtime_end),
+                              ad.blurb)
+        table.reformat_column(2, width=8)
+        table.reformat_column(1, width=18)
+        table.reformat_column(0, width=20)
+        self.msg(table)
+
+    def display_all_matches(self):
+        my_ad = self.caller.dompc.wanted_ad
+        ads = WantedAd.objects.filter(active=True)
+        table = evtable.EvTable(
+            "Name/Gender/Age",
+            "Matches",
+            "Times",
+            "Blurb",
+            border="cells",
+            width=78)
+        for ad in ads:
+            match = ad.match(self.caller.dompc)
```

Comment:
Yeah, the O(N) queries in ad.match could actually be pretty troublesome - even something like 50 wanted adds doing queries on search tags and clues might be noticeable. What I'd probably do is turn it into a manager method of WantedAd, like `WantedAd.objects.find_matches(self.caller.dompc)`. But again, if that's not something you want to mess with, don't worry about it, it wouldn't take me long to figure something out.